lib/doctest.mjs

```diff
@@ -22,35 +22,41 @@ export default async function(path, options) {
       common.sanitizeFileContents (
         await util.promisify (fs.readFile) (path, 'utf8')
       )
-    )
+    ),
+    options.logFunction
   );
 
   if (options.print) {
     console.log (source.replace (/\n$/, ''));
     return [];
   } else if (options.silent) {
-    return evaluate (source, path);
+    return evaluate (options, source, path);
   } else {
     console.log ('running doctests in ' + path + '...');
-    return (evaluate (source, path)).then (function(results) {
+    return (evaluate (options, source, path)).then (function(results) {
       doctest.log (results);
       return results;
     });
   }
 }
 
-function wrap(source) {
+function wrap(source, logFunction) {
   return common.unlines ([
     'export const __doctest = {',
     '  queue: [],',
-    '  enqueue: function(io) { this.queue.push(io); }',
+    '  enqueue: function(io) { this.queue.push(io); },',
+    '  logMediator: {emit: function(){}}',
     '};',
-    '',
-    source
-  ]);
+    ''
+  ]) + (logFunction != null ? common.unlines ([
+    'const ' + logFunction + ' = channel => value => {',
+    '  __doctest.logMediator.emit ({channel, value});',
+    '};',
```

Comment:
I think this is our first use of arrow functions. Should we also use `() => {}` in place of `function() {}` a few lines above?

Reply:
Oh. Perhaps I should use a regular function.